packages/engine/solana/crawler.ts

```diff
@@ -25,33 +28,58 @@ export const solanaEngineRunner: EngineRunner<Connection> = {
 					address: key._id,
 				}),
 			);
-		}
 
-		const tokenChunks = await Promise.all(tokenPromises);
-		const tokenDocuments = flatten(tokenChunks);
-		const makeId = (i: TokenDocument) => `${i.network}#${i.account.mint}`;
-
-		try {
-			const { tokensByAddress } = await qlClient.request<
-				{ tokensByAddress: TokenInfo[] },
-				{ addresses: string[] }
-			>(queries.tokensByAddress, {
-				addresses: tokenDocuments.map(makeId),
-			});
-
-			const quoteMap = tokensByAddress.reduce((a, i) => {
-				a[i.address as string] = i;
-				return a;
-			}, {} as Record<string, TokenInfo>);
-
-			for (const i of tokenDocuments) {
-				i.account.quotes = quoteMap[makeId(i)].quotes;
-			}
-		} catch (_) {
-			console.log('cannot fetch solana token price');
+			collectiblePromises.push(
+				solanaCollectiblesByAddress({
+					endpoint,
+					connection,
+					address: key._id,
+				}),
+			);
 		}
 
-		tokenActions.setItems(tokenDocuments);
+		const promises = [];
+		promises.push(
+			Promise.all(tokenPromises).then(async (tokenChunks) => {
+				const tokenDocuments = flatten(tokenChunks);
+				const makeId = (i: TokenDocument) => `${i.network}#${i.account.mint}`;
+
+				try {
+					const { tokensByAddress } = await qlClient.request<
+						{ tokensByAddress: TokenInfo[] },
+						{ addresses: string[] }
+					>(queries.tokensByAddress, {
+						addresses: tokenDocuments.map(makeId),
+					});
+
+					const quoteMap = tokensByAddress.reduce((a, i) => {
+						a[i.address as string] = i;
+						return a;
+					}, {} as Record<string, TokenInfo>);
+
+					for (const i of tokenDocuments) {
+						i.account.quotes = quoteMap[makeId(i)].quotes;
+					}
+				} catch (_) {
+					console.log('cannot fetch solana token price');
+				}
+
+				tokenActions.setItems(tokenDocuments);
+			}),
+		);
+
+		promises.push(
+			Promise.all(collectiblePromises).then(async (results) => {
+				collectibleActions.setCollections(
+					flatten(results.map((result) => result.collections)),
+				);
+				collectibleActions.setCollectibles(
+					flatten(results.map((result) => result.collectibles)),
+				);
+			}),
+		);
```

Comment:
@ltminhthu here is where we initialize all nfts with their collections. update state inside engine